lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -538,6 +537,15 @@ def create_coefficients_cube(
                 time_coord = historic_forecast.coord("time").copy(time_point)
                 aux_coords_and_dims.append((time_coord, None))
 
+        # Create x and y coordinates
+        for axis in ["x", "y"]:
+            coord_point = np.median(historic_forecast.coord(axis=axis).points)
+            coord_bounds = [historic_forecast.coord(axis=axis).points[0],
+                            historic_forecast.coord(axis=axis).points[-1]]
```

Comment:
The code is doing the right thing, but it would be nicer if the coord points that are accessed three times were extracted first:
historic_coord_points = historic_forecast.coord(axis=axis).points
coord_point = np.median(historic_coord_points)
etc.

The same argument holds in the unit-tests too along with the observation that the loop over x and y in the code is not as elegant in the unit-test.

Reply:
I've amended the code as suggested.